src/main/java/org/jfrog/hudson/ArtifactoryServer.java

```diff
@@ -414,4 +430,35 @@ public ConverterImpl(XStream2 xstream) {
             super(xstream);
         }
     }
+
+    @Extension
+    public static class DescriptorImpl extends Descriptor<ArtifactoryServer> {
+
+        public static final int DEFAULT_DEPLOYMENT_THREADS_NUMBER = 3;
+        public static final int DEFAULT_CONNECTION_RETRY_NUMBER = 3;
+        public static final int DEFAULT_CONNECTION_TIMEOUT = 300;    // 5 Minutes
+
+        @Override
+        @NonNull
+        public String getDisplayName() {
+            return "Artifactory Server";
+        }
+
+        public ListBoxModel doFillConnectionRetryItems() {
```

Comment:
Or just use `<f:number clazz="positive-number" min="0" step="1" />` in the Jelly file.

Reply:
Fixed